.github/workflows/build_shadow.yml

```diff
@@ -0,0 +1,24 @@
+# Syntax reference:
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions
+
+name: Test
```

Comment:
This is what appears on the badge. Should we use the plural form so it reads `Tests: Passing`?

An aside: [here are some other badges that are cool](https://github.com/badges/shields) :)

Reply:
Done.

The custom badges are neat. It's not clear to me how to easily wire these up to the test results. Maybe something to poke at later, but I'd rather get this in first :).

Reply:
For the tests it seems safer to use an ignore-list than an include-list. I set it to ignore markdown files, which seems pretty safe.

I changed the python action to only trigger and setup and *.py.

Unfortunately I don't see a way to set up file filters for multiple triggers (push and pull_request) without duplicating. (I suppose we could use another tool to generate these files and check in the generated output when we change the source, but seems a bit overkill for now).

Reply:
Tried using YAML anchors to deduplicate the filter paths, but github doesn't support them: https://github.com/sporksmith/shadow/runs/417939170?check_suite_focus=true